src/Mockery/Type/MockDynamicReturnTypeExtension.php

```diff
@@ -41,15 +41,36 @@ public function getTypeFromStaticMethodCall(
 			return $defaultReturnType;
 		}
 
-		$classType = $scope->getType($methodCall->args[0]->value);
-		if (!$classType instanceof ConstantStringType) {
-			return $defaultReturnType;
+		$types = [$defaultReturnType];
+		foreach ($methodCall->args as $arg) {
+			$classType = $scope->getType($arg->value);
+			if (!$classType instanceof ConstantStringType) {
+				continue;
+			}
+
+			$value = $classType->getValue();
+			if (substr($value, 0, 6) === 'alias:') {
+				$value = substr($value, 6);
+			}
+			if (substr($value, 0, 9) === 'overload:') {
+				$value = substr($value, 9);
+			}
+			if (substr($value, -1) === ']' && strpos($value, '[') !== false) {
```

Comment:
There's something about parsing `[]` but I don't see a corresponding test? Otherwise good job! 👍

Reply:
Good catch! Added it